### PR TITLE
fix requestID using consumerID && loopclosure issue && add old style topic support

### PR DESCRIPTION
**Pull request by archfish**
### Modifications

* fix requestID using consumerID
* fix loopclosure issue
* add old style topic support

### CODE REVIEW

File: pulsar/internal/topic_name.go [hunk 1]
```diff
@@ -44,9 +44,10 @@ func ParseTopicName(topic string) (*TopicName, error) {
 	if !strings.Contains(topic, "://") {
 		// The short topic name can be:
 		// - <topic>
-		// - <property>/<namespace>/<topic>
+		// - <tenant>/<namespace>/<topic>
+		// - <tenant>/<cluster>/<namespace>/<topic>
 		parts := strings.Split(topic, "/")
-		if len(parts) == 3 {
+		if len(parts) == 3 || len(parts) == 4 {
```
Comment (cckellogg) [inline pulsar/internal/topic_name.go]: Overall looks good. Let's add a test case for this.

Comment (archfish) [reply pulsar/internal/topic_name.go]:
```log
Running tool: /usr/local/bin/go test -timeout 30s github.com/apache/pulsar-client-go/pulsar/internal -run ^(TestParseTopicName)$

ok  	github.com/apache/pulsar-client-go/pulsar/internal	0.559s
Success: Tests passed.
```

File: pulsar/internal/topic_name_test.go [hunk 2]
```diff
@@ -43,6 +43,12 @@ func TestParseTopicName(t *testing.T) {
 	assert.Equal(t, "my-tenant/my-namespace", topic.Namespace)
 	assert.Equal(t, -1, topic.Partition)
 
+	topic, err = ParseTopicName("my-tenant/my-cluster/my-namespace/my-topic")
```
Comment (cckellogg) [inline pulsar/internal/topic_name_test.go]: nitpick move this to the bottom where the other V1 topic name tests are.

Comment (archfish) [reply pulsar/internal/topic_name_test.go]: fixed

**Pull request merged**